Add example for Azure OIDC creds in insights #27703 (#15297)
pulumi/docs: content/docs/insights/accounts.md

### Azure

> The Azure scanner for Pulumi Cloud requires access to your Azure account. This access can be granted by creating an ESC environment that, when opened, produces valid credentials to use the Pulumi Azure provider.

Reviewer: ```suggestion
The Azure scanner for Pulumi Cloud requires access to your Azure subscription. This access can be granted by creating an ESC environment that, when opened, produces valid credentials for the Pulumi Azure Native provider to consume.
```

> {{< notes type="info" >}}

Reviewer: 👍

  We recommend using OpenID Connect (OIDC) for passwordless authentication with Azure. This method is more secure than using client secrets as it eliminates the need to store and manage long-lived credentials.
{{< /notes >}}

#### Option 1: OIDC authentication (recommended)

1. Create a Microsoft Entra application and configure federated credentials:
   * Follow the steps in [Configuring OpenID Connect for Azure](/docs/esc/environments/configuring-oidc/azure/)
   * When configuring the federated credential:

>      * **Audience**: `azure:{your-org-name}`

Reviewer: ```suggestion
     * **Audience**: `azure:<your-pulumi-org-name>`
```

>      * **Subject identifier**: `pulumi:environments:org:{your-org-name}:env:{project-name}/{environment-name}`

Reviewer: ```suggestion
     * **Subject identifier**: `pulumi:environments:org:<your-pulumi-org-name>:env:<esc-project-name>/<esc-environment-name>`
```

For greater clarity and consistency with our convention of using `<` and `>` as delimiters (such as it is).

   * Note the Application (client) ID, Directory (tenant) ID, and Subscription ID

> 2. Use the following ESC configuration:
> 
> ```yaml
> values:
>   azure:
>     login:
>       fn::open::azure-login:
>         clientId: <YOUR_CLIENT_ID>
>         tenantId: <YOUR_TENANT_ID>
>         subscriptionId: <YOUR_SUBSCRIPTION_ID>
>         oidc: true
>   environmentVariables:
>     ARM_USE_OIDC: 'true'
>     ARM_CLIENT_ID: ${azure.login.clientId}
>     ARM_TENANT_ID: ${azure.login.tenantId}
>     ARM_OIDC_TOKEN: ${azure.login.oidc.token}
>     ARM_SUBSCRIPTION_ID: ${azure.login.subscriptionId}
> ```

Reviewer: ```suggestion
1. Use the following ESC configuration:

    ```yaml
    values:
      azure:
        login:
          fn::open::azure-login:
            clientId: <YOUR_CLIENT_ID>
            tenantId: <YOUR_TENANT_ID>
            subscriptionId: <YOUR_SUBSCRIPTION_ID>
            oidc: true
      environmentVariables:
        ARM_USE_OIDC: 'true'
        ARM_CLIENT_ID: ${azure.login.clientId}
        ARM_TENANT_ID: ${azure.login.tenantId}
        ARM_OIDC_TOKEN: ${azure.login.oidc.token}
        ARM_SUBSCRIPTION_ID: ${azure.login.subscriptionId}
    ```
```

Reviewer: Codefences have to be indented 4 spaces to appear under a list item.

#### Option 2: Client secret authentication

1. Create a Service Principal in Azure, then generate the following values:
   * **clientId** (also called **appId** in the Azure UI)
   * **tenantId**
   * **subscriptionId**
   * **clientSecret** (also called **password** in the Azure UI)

> 2. Use the following ESC configuration:

Reviewer: ```suggestion
1. Use the following ESC configuration:
```

Ordered lists will autonumber. Using `1.` everywhere will make future diffs easier to read.

Reviewer: Need to make sure this won't mess up the numbering further down, but looks like this is the end of the list.
